packages/core/src/agent/index.ts

```diff
@@ -54,6 +55,16 @@ function resolveMaybePromise<T, R = void>(value: T | Promise<T>, cb: (value: T)
   return cb(value);
 }
 
+// Helper to resolve threadId from args (supports both new and old API)
+function resolveThreadIdFromArgs(args: any): (Partial<StorageThreadType> & { id: string }) | undefined {
```

Comment:
I think we can likely type this as something like `Parameters<(typeof agent)["generate"]>[0]` instead of any